app/Jobs/ProcessNewVideo.php

```diff
@@ -215,60 +228,84 @@ protected function getVideoDimensions($url)
+        $p->run();
+        if ($p->getExitCode() !== 0) {
+            throw new Exception("Process was terminated with code {$p->getExitCode()}");
         }
 
-        $buffer = $this->ffmpegVideo->frame(TimeCode::fromSeconds($time))
-            ->save(null, false, true);
-
-        return VipsImage::thumbnail_buffer($buffer, $width, ['height' => $height])
-            ->writeToBuffer(".{$format}", [
-                'Q' => 85,
-                'strip' => true,
-            ]);
     }
 
-    /**
-     * Get the times at which thumbnails should be sampled.
-     *
-     * @param float $duration Video duration.
-     *
-     * @return array
-     */
-    protected function getThumbnailTimes($duration)
+    public function generateVideoThumbnails($disk, $fragment, $tmpDir)
     {
-        $count = config('videos.thumbnail_count');
 
-        if ($count <= 1) {
-            return [$duration / 2];
-        }
+        // Config for normal thumbs
+        $format = config('thumbnails.format');
+        $thumbCount = config('videos.thumbnail_count');
+        $width = config('thumbnails.width');
+        $height = config('thumbnails.height');
 
-        // Start from 0.5 and stop at $duration - 0.5 because FFMpeg sometimes does not
-        // extract frames from a time code that is equal to 0 or $duration.
-        $step = ($duration - 1) / floatval($count - 1);
-        $start = 0.5;
-        $end = $duration - 0.5;
-        $range = range($start, $end, $step);
+        // Config for sprite thumbs
+        $thumbnailsPerSprite = config('videos.sprites_thumbnails_per_sprite');
+        $thumbnailsPerRow = sqrt($thumbnailsPerSprite);
+        $thumbnailWidth = config('videos.sprites_thumbnail_width');
+        $thumbnailHeight = config('videos.sprites_thumbnail_height');
+        $spriteFormat = config('videos.sprites_format');
+
+        $files = File::glob($tmpDir . "/*.{$format}");
+        $nbrFiles = count($files);
+        $steps = $nbrFiles >= $thumbCount ? floor($nbrFiles / $thumbCount) : 1;
+
+        $thumbnails = [];
+        $thumbCounter = 0;
+        $spriteCounter = 0;
+        foreach ($files as $i => $file) {
+            if ($i === intval($steps*$thumbCounter) && $thumbCounter < $thumbCount) {
+                $thumbnail = VipsImage::thumbnail($file, $width, ['height' => $height]);
+                $bufferedThumb = $thumbnail->writeToBuffer(".{$format}", [
+                    'Q' => 85,
+                    'strip' => true,
+                ]);
+                $disk->put("{$fragment}/{$thumbCounter}.{$format}", $bufferedThumb);
+                $thumbCounter += 1;
+            }
 
-        // Sometimes there is one entry too few due to rounding errors.
-        if (count($range) < $count) {
-            $range[] = $end;
-        }
+            if (count($thumbnails) < $thumbnailsPerSprite) {
+                $thumbnails[] = VipsImage::thumbnail($file, $thumbnailWidth, ['height' => $thumbnailHeight, 'size' => 'force']);
```

Comment:
I think the `'size' => 'force'` argument still stretches the thumbnails in an unintended way. Compare the sprite thumbnail (top) to the regular thumbnail (bottom):
![0-sprite](https://github.com/user-attachments/assets/8fe493ca-ac49-400a-8c98-06402140bb59)
![0](https://github.com/user-attachments/assets/d03b0638-be2e-45e6-a071-e3b95008f5b0)

The sprite preview should also use the correct aspect ratio like the regular thumbnail and adjust the preview element size accordingly.